server/tests/chat/test_datadog_metric_stats.py

```
@@ -0,0 +1,408 @@
+                            [1.0, float("nan"), 3.0, 2.0])
+    assert row["max"] == 3.0
+    assert row["nulls"] == 1
+    assert row["points"] == 4
+
+
+def test_non_finite_points_never_reach_json():
+    """json.dumps emits bare NaN/Infinity, which is invalid JSON and would fail
+    the agent's parse of the whole tool result."""
+    row = _summarize_series(["x"], None, [1.0, float("inf"), float("-inf"), 2.0])
+    serialized = json.dumps(row)
+    assert "Infinity" not in serialized and "NaN" not in serialized
```

Ruff (0.16.0)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)